conans/client/tools/oss.py

```diff
@@ -48,17 +48,13 @@ def detected_os():
 def detected_architecture():
     # FIXME: Very weak check but not very common to run conan in other architectures
     machine = platform.machine()
+    os_info = OSInfo()
 
-    if OSInfo().is_solaris:
-        # under intel solaris, platform.machine()=='i86pc' so we need to handle it early to suport 64-bit
-        processor = platform.processor()
-        kernel_bitness, elf = platform.architecture()
-        if "sparc" in processor:
-            return "sparcv9" if kernel_bitness == "64bit" else "sparc"
-        elif "i386" in processor:
-            return "x86_64" if kernel_bitness == "64bit" else "x86"
- 
-    if "ppc64le" in machine:
+    if os_info.is_solaris:
+        return OSInfo.get_solaris_architecture()
+    elif os_info.is_aix:
+        return OSInfo.get_aix_architecture()
```

Comment:
Moving this condition here is not just a refactor. Probably this is the right place for this condition.

Reply:
I know, I changed the title of the PR. I took it into account and I think this is ok and more clear to move it here.

Reply:
The changelog is still irrelevant for the users. The slightly changed behavior shouldn't affect anyone.